tests/console/nftables.pm

```diff
@@ -0,0 +1,113 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: nftables
+# Summary: Test nftables with firewalld
+# Maintainer: Zaoliang Luo <zluo@suse.de>
+
+use strict;
+use warnings;
+use base "consoletest";
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # check nftables, stop iptables and start firewalld by need
+    my $logs = "/tmp/log";
+
+    #check that nftables has been already installed
+    assert_script_run('echo -e "____ nftables version and rules ____\n\n" > ' . $logs);
+    assert_script_run("rpm -qa nftables |& tee -a $logs");
+    assert_script_run("if systemctl is-active -q iptables; then systemctl stop iptables; fi");
+    systemctl("restart firewalld");
+
+    # check that directory of nftables and create a customer rule 'tcp-mss-clamp'
+    assert_script_run('echo -e "____ /etc/nftables contains rules ____\n" >> ' . $logs);
+    assert_script_run("ls /etc/nftables |& tee -a $logs");
+
+    script_run('cat > tcp-mss-clamp <<EOF
+#  nft list chain inet firewalld filter_FWDO_public_allow
+table inet firewalld {
+        chain filter_FWDO_public_allow {
+                tcp flags syn tcp option maxseg size set rt mtu
+        }
+}
+
+EOF
+true');
+
+    # run and check customer rule, add firewall rule and restart firewalld
+    assert_script_run("nft flush ruleset");
+    assert_script_run("nft -f tcp-mss-clamp");
+    assert_script_run("nft list chain inet firewalld filter_FWDO_public_allow");
+    assert_script_run("nft list tables");
```

Comment:
These tests depend on a set of rules already being present, but none of them are actually checking that the output is what should be expected